scipy/optimize/_removeRedundancy.py

```diff
@@ -0,0 +1,208 @@
+    """
+    Returns the index of the densest row of A. Ignores rows that are not
+    eligible for consideration.
+
+    Parameters
+    ----------
+    A : 2-D array
+        An array representing a matrix
+    eligibleRows : 1-D logical array
+        Values indicate whether the corresponding row of A is eligible 
+        to be considered
+        
+    Returns
+    -------
+    i_densest : int
+        Index of the densest row in A eligible for consideration
+        
+    """
+    rowCounts = _rowCount(A)
+    return np.argmax(rowCounts*eligibleRows)
+    
+def _removeZeroRows(A,b):
+    """
+    Eliminates trivial equations from system of equations defined by Ax = b
+   and identifies trivial infeasibilities 
+
+    Parameters
+    ----------
+    A : 2-D array
+        An array representing the left-hand side of a system of equations
+    b : 1-D array
+        An array representing the right-hand side of a system of equations
+        
+    Returns
+    -------
+    A : 2-D array
+        An array representing the left-hand side of a system of equations
+    b : 1-D array
+        An array representing the right-hand side of a system of equations
+    status: int
+        An integer indicating the status of the removal operation
+        0: No infeasibility identified
+        2: Trivially infeasible
+        
+    """
+    status = 0
+    i_zero = _rowCount(A) == 0
+    A = A[np.logical_not(i_zero),:]
+    if not(np.allclose(b[i_zero],0)):
+        status = 2    
+    b = b[np.logical_not(i_zero)]
+    return A,b,status
+
+def _removeRedundancy(A,b):
+    """
+    Eliminates redundant equations from system of equations defined by Ax = b
+    and identifies infeasibilities.
+
+    Parameters
+    ----------
+    A : 2-D array
+        An array representing the left-hand side of a system of equations
+    b : 1-D array
+        An array representing the right-hand side of a system of equations
+        
+    Returns
+    -------
+    A : 2-D array
+        An array representing the left-hand side of a system of equations
+    b : 1-D array
+        An array representing the right-hand side of a system of equations
+    status: int
+        An integer indicating the status of the system
+        0: No infeasibility identified
+        2: Trivially infeasible
+        
+    """
+
+    A,b,status = _removeZeroRows(A,b)
+
+    if status != 0:
+        return A,b,status
+        
+    U,s,Vh = sp.linalg.svd(A)
+    eps = np.finfo(float).eps
+    tol = s.max() * max(A.shape) * eps
+    
+    m,n = A.shape
+    s_min = s[-1] if m  <= n else 0 
+    
+    while abs(s_min) < tol:
+        v = U[:,-1]
+        eligibleRows = np.abs(v) > tol*10e6 # rows need to be represented in significant amount
+        if not np.any(eligibleRows) or np.any(np.abs(v.dot(A)) > tol):
+            status = 4
+            break
+#            print "Due to numerical issues, redundant equality constraints cannot be removed automatically."
```

Comment:
I turned these into messages corresponding with the `status`. There are a few other pieces of recently commented out code corresponding with a change from lambda functions to regular functions that I will remove for the final version.